src/main/java/hudson/plugins/scm_sync_configuration/strategies/model/PatternsEntityMatcher.java

```diff
@@ -22,12 +27,33 @@ public boolean matches(Saveable saveable, File file) {
 		if (file == null) {
 			return false;
 		}
-		String filePathRelativeToHudsonRoot = JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file);
-        AntPathMatcher matcher = new AntPathMatcher();
-        for(String pattern : includesPatterns) {
-            if(matcher.match(pattern, filePathRelativeToHudsonRoot)){
-                return true;
-            }
+		return matches(saveable, JenkinsFilesHelper.buildPathRelativeToHudsonRoot(file), file.isDirectory());
+	}
+
+	public boolean matches(Saveable saveable, String pathRelativeToRoot, boolean isDirectory) {
+		if (pathRelativeToRoot != null) {
+			// Guard our own SCM workspace and the war directory. User-defined includes might inadvertently include those if they start with * or **!
+			if (pathRelativeToRoot.equals(SCM_WORKING_DIRECTORY) || pathRelativeToRoot.startsWith(SCM_WORKING_DIRECTORY + '/')) {
+				return false;
+			} else if (pathRelativeToRoot.equals(WAR_DIRECTORY) || pathRelativeToRoot.startsWith(WAR_DIRECTORY + '/')) {
+				return false;
+			}
+	        AntPathMatcher matcher = new AntPathMatcher();
+	        String directoryName = null;
+	        for (String pattern : includesPatterns) {
+	            if (matcher.match(pattern, pathRelativeToRoot)) {
+	                return true;
+	            } else if (isDirectory) {
+	            	// pathRelativeFromRoot is be a directory, and the pattern end in a file name. In this case, we must claim a match.
```

Comment:
Hi. I must say I don't quite understand the change you've referenced: I would have thought that getAbsolutePath() was a pure path operation. If we used getCanonicalPath() then that test might indeed detect symlinks, but I doubt it does with getAbsolutePath().

But anyway, check [line 20 in JenkinsFilesHelper](https://github.com/jenkinsci/scm-sync-configuration-plugin/pull/31/files#diff-2fe44d7c1568160f1fa01c43dc8a2098R20) and the calls to this matches() operation here. I think that's the same test you did, just in a different place.

Also, with these changes the plugin will not synchronize build history or nextBuildNumber by default. A user would have to include those explicitly. Maybe you could explain better over at PR #28 what exactly the setup is and what exceptions were logged, and what the plugin tried to synchronize. (Note that the plugin 0.0.8 had a bug (JENKINS-16378, or issue #8 here) that mistakenly started synchronizing everything in a job's directory when a job was renamed. That should be fixed here with [commit 28d4aee2](https://github.com/tomaswolf/scm-sync-configuration-plugin/commit/28d4aee27703e1b02ad2989f6d5288fe47644ea6) in this PR.) Finally, PR #28 appears to be based on grzegorz7's work on branch specification, which greatly confuses matters.

But I see that something went wrong with indentation. Probably my Eclipse used tabs to indent. I'll have to fix that so that the diff gets more readable.